src/lib/components/ui/input/date-input.svelte

```diff
@@ -1,8 +1,286 @@
+			currentMonth === currentDate.getMonth() &&
+			currentYear === currentDate.getFullYear()
+		)
+	}
+	function formatDate(date: Date) {
+		const year = date.getFullYear()
+		const month = String(date.getMonth() + 1).padStart(2, '0')
+		const day = String(date.getDate()).padStart(2, '0')
+		return `${year}-${month}-${day}`
+	}
+
+	let daysInMonth = $derived(new Date(currentYear, currentMonth + 1, 0).getDate())
+	let firstDayOfMonth = $derived((new Date(currentYear, currentMonth, 1).getDay() + 6) % 7)
+	let prevMonthDays = $derived(new Date(currentYear, currentMonth, 0).getDate())
+	let totalCells = 42
+	let calendarDays = $derived(
+		Array.from({ length: totalCells }, (_, i) => {
+			const dayNumber = i - firstDayOfMonth + 1
+			if (dayNumber <= 0) {
+				return {
+					date: prevMonthDays + dayNumber,
+					type: 'prev',
+				}
+			} else if (dayNumber > daysInMonth) {
+				return {
+					date: dayNumber - daysInMonth,
+					type: 'next',
+				}
+			} else {
+				return {
+					date: dayNumber,
+					type: 'current',
+				}
+			}
+		}),
+	)
+	let size: 16 | 24 | 32 = $derived(dimension === 'large' ? 32 : dimension === 'small' ? 16 : 24)
+	let showDatePicker = $state(false)
+	let variant: Variant = $derived(
+		dimension === 'large' ? 'large' : dimension === 'small' ? 'small' : 'default',
+	)
 </script>
 
-<Input type="date" {...restProps} />
+{#snippet buttons()}
+	<Button
+		{dimension}
+		{disabled}
+		variant="secondary"
+		onclick={() => {
+			showDatePicker = !showDatePicker
+		}}
+	>
+		<Calendar {size} />
+	</Button>
+{/snippet}
+
+<div class="calendar-root {dimension}">
+	<Input
+		{dimension}
+		{disabled}
+		{...restProps}
+		value={formatDate(selectedDate)}
+		{buttons}
+		type="date"
+	/>
+	<div class="date-picker" class:showDatePicker>
+		<div class="header">
+			<div class="month">
+				<Button {dimension} variant="ghost" onclick={() => changeMonth(-1)}
+					><ChevronLeft {size} /></Button
+				>
+				<div class="current-month">
+					<Typography {variant} bold>{months[currentMonth]} {currentYear}</Typography>
+					<Button {dimension} variant="ghost" onclick={toggleYearPicker}>
+						{#if showYearPicker}
+							<CaretUp {size} />
+						{:else}
+							<CaretDown {size} />
+						{/if}
+					</Button>
+				</div>
+				<Button {dimension} variant="ghost" onclick={() => changeMonth(1)}
+					><ChevronRight {size} /></Button
+				>
+			</div>
+			{#if !showYearPicker}
+				<div class="days">
+					{#each days as day}
+						<Typography {variant}>{day}</Typography>
+					{/each}
+				</div>
+			{/if}
+		</div>
+
+		<Divider class="no-margin" />
+
+		{#if showYearPicker}
+			<div class="year-picker">
+				<!-- eslint-disable-next-line @typescript-eslint/no-unused-vars -->
+				{#each Array(131) as _, i}
+					<Button
+						{dimension}
+						variant="ghost"
+						active={isSelected(1970 + i) || isCurrentYear(1970 + i)}
+						onclick={() => changeYear(1970 + i)}>{1970 + i}</Button
+					>
+				{/each}
```

Comment:
Generally it would be good to generate these as user scrolls but we can live with this in v1